core/src/main/java/com/sap/ai/sdk/core/AiCoreService.java

```diff
@@ -0,0 +1,95 @@
+package com.sap.ai.sdk.core;
+
+import static com.sap.ai.sdk.core.AiCoreServiceWithDeployment.getDeploymentId;
+import static com.sap.ai.sdk.core.AiCoreServiceWithDeployment.isDeploymentOfModel;
+
+import com.sap.ai.sdk.core.client.model.AiDeployment;
+import com.sap.cloud.sdk.cloudplatform.connectivity.Destination;
+import com.sap.cloud.sdk.cloudplatform.connectivity.exception.DestinationAccessException;
+import com.sap.cloud.sdk.cloudplatform.connectivity.exception.DestinationNotFoundException;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/** Connectivity convenience methods for AI Core. */
+@Slf4j
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
+public class AiCoreService implements AiCoreServiceStub {
```

Comment:
What made you change from `AiClient` _(or `AiCoreClient`)_ to `AiCoreService`.
The service is in BTP, we only offer a client.

Reply:
Yes, `AiCoreService` represents and is used for the entire AI Core service, while we have several clients (orchestration, openAi, DeploymentAPI etc.)